### ISSUE TITLE
`no-undef` false positive

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#5420]
From https://twitter.com/ymyzk/status/703426419481661440 (Japanese)

`no-undef` is warning `new.target` meta property.

```
> eslint -v
v2.2.0
> echo "class A { constructor() { new.target; } }" | eslint --stdin --env es6 --rule no-undef:2 --no-eslintrc

<text>
  1:27  error  'new' is not defined     no-undef
  1:31  error  'target' is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)
```

This seems to be a bug of `escope`.
`escope` has been creating references of `new` and `target` identifiers.
I will try to fix this.

**Comment by rozzzly** [comment]
Im also getting it on `const derp = Symbol();`

**Comment by mysticatea** [comment]
@rozzzly Can you open a new issue? This issue is about `new.target` meta property.

**Comment by rozzzly** [comment]
Well I found that I just had to edit:
#####  _**`.eslintrc`**_

```javascript
  // ...
  "env": {
    "es6": true,
    "browser": true,
    "node": true
  },
  // ...
```
